proto/scenario.proto

```diff
@@ -0,0 +1,116 @@
+syntax = "proto3";
+package proto;
+
+message Scenario {
+  Load load = 1;
+  Cluster cluster = 2;
+  Workload workload = 3;
+  repeated Plugin plugin = 4;
+}
+
+message Load {
+  oneof arrival_rate_oneof {
+    LoadConstant constant = 1;
+    LoadPeriodic periodic = 3;
+    LoadCustom custom = 3;
+  }
+  oneof request_weight_oneof {
+    RequestConstant constant = 4;
+  }
+}
+
+message LoadConstant {
+  int32 arrivals_per_second = 1;
+}
+
+message LoadPeriodic {
+  oneof shape_oneof {
+    ShapeSine sine = 1;
+    ShapeAsdr asdr = 2;
+  }
+}
+
+message ShapeSine {
+  LoadInterval interval = 1;
+}
+
+message ShapeAsdr {
+  LoadInterval initial = 1;
+  LoadInterval attack = 2;
+  LoadInterval sustain = 3;
+  LoadInterval decay = 4;
+  LoadInterval release = 5;
+}
+
+message LoadCustom {
+  repeated LoadInterval interval = 1;
+}
+
+message LoadInterval {
+  int32 width_milliseconds = 1;
+  int32 amplitude_aps = 2;
```

Comment:
APS? Is this a typo for QPS (the google shorthand as I understand it)?

If so, should it be in per-second units? I can see that it's easier to approach from how folks usually express load, but it also means that we'll be internally doing a `int32 / int32` to derive the correct value and will need to be careful of overflows into (I presume) `float64`.

I feel like it would be easier to give a total number of load units (requests, queries). That does put it back on the user to perform a calculation but it means that they don't have to perform any calculations to recover the total number.

Reply:
I was calling it "arrivals per second" because I want to remain open to pub-sub use-cases.  But "operations per second" would also work.  Would that be more intuitive?

(QPS - "queries per second" is probably not the best choice)

Reply:
No, I think you're right. Arrivals/sec is a better unit. Perhaps rename as `amplitude_arrivals_per_second`, `arrivals_per_sec` or similar.